lang/go/bindings/examples/sampling/sampling.jsnoc

```diff
@@ -0,0 +1,159 @@
+    //    description   - (O) the description, 'desc' can be used as well
+    "lports": {
+        "eno12399:0": {
+            "pmd": "net_af_xdp",
+            "qid": 0,
+            "umem": "umem0",
+            "region": 0,
+            "description": "LAN 0 port"
+        }
+    },
+
+    // (O) Define the lcore groups for each thread to run
+    //     Can be integers or a string for a range of lcores
+    //     e.g. [10], [10-14,16], [10-12, 14-15, 17-18, 20]
+    // Names of a lcore group and its lcores assigned to the group.
+    // The initial group is for the main thread of the application.
+    // The default group is special and is used if a thread if not assigned to a group.
+    "lcore-groups": {
+        "initial": [0],
+        "group0": [1],
+        "group1": [2],
+        "default": ["0"]
+    },
+
+    // (O) Set of common options application defined.
+    //     The Key can be any string and value can be boolean, string, array or integer
+    //     An array must contain only a single value type, boolean, integer, string and
+    //     can't be a nested array.
+    //   pkt_api    - (O) Set the type of packet API xskdev, xdpdev or pktdev
+    //   no-metrics - (O) Disable metrics gathering and thread
+    //   no-restapi - (O) Disable RestAPI support
+    //   cli        - (O) Enable/Disable CLI supported
+    //   mode       - (O) Mode type [drop | rx-only], tx-only, [lb | loopback], fwd, acl-strict, acl-permissive
+    //   no-color   - (O) Enable/Disable vt100 colors
+    //   burst-size - (O) Burst size for application, if zero or not defined use default 128, max 256.
+    //   uds_path   - (O) Path to unix domain socket to get xsk map fd
+    "options": {
+        "no-metrics": false,
+        "no-restapi": false,
+        "cli": true,
+        "mode": "l3fwd",
+	"l3fwd-fib-rules": [
+             "198.18.0.0/24,70:b5:e8:f0:31:30,1",
+             "198.18.1.0/24,b4:96:91:ba:0a:de,0"
+	],
+        "no-color": false,
+        "burst-size": 64
+    },
+
+    // List of threads to start and information for that thread. Application can start
+    // it's own threads for any reason and are not required to be configured by this file.
+    //
+    //   Key/Val   - (R) A unique thread name.
+    //                   The format is <type-string>[:<identifier>] the ':' and identifier
+    //                   are optional if all thread names are unique
+    //      group  - (O) The lcore-group this thread belongs to. The
+    //      lports - (O) The list of lports assigned to this thread and can not shared lports.
+    //      description | desc - (O) The description
+    "threads": {
+        "main": {
+            "group": "initial",
+            "description": "Main Thread"
+        },
+        "fwd:0": {
+            "group": "group0",
+            "lports": ["eno12399:0"],
+            "description": "Forward thread 0"
+        },
+        "fwd:1": {
+            "group": "group1",
+            "lports": ["eno12399:0"],
```

Comment:
Having two threads manage the same port should not be done. The transmit side of CNDP has protection from two or more threads accessing the same netdev/qid, but the receive side does not have that protection. Normally we have only one thread doing receives on a given netdev/qid, but that thread can send to any other netdev/qid port to allow forwarding of packets.